src/debugger/cli-debugger.c

```diff
@@ -717,9 +784,17 @@ static void _listWatchpoints(struct CLIDebugger* debugger, struct CLIDebugVector
 	for (i = 0; i < mWatchpointListSize(&watchpoints); ++i) {
 		struct mWatchpoint* watchpoint = mWatchpointListGetPointer(&watchpoints, i);
 		if (watchpoint->segment >= 0) {
-			debugger->backend->printf(debugger->backend, "%" PRIz "i: %02X:%X\n", watchpoint->id, watchpoint->segment, watchpoint->address);
+			if (watchpoint->maxAddress == watchpoint->minAddress + 1) {
+				debugger->backend->printf(debugger->backend, "%" PRIz "i: %02X:%X\n", watchpoint->id, watchpoint->segment, watchpoint->minAddress);
+			} else {
+				debugger->backend->printf(debugger->backend, "%" PRIz "i: [%02X:%X,%02X:%X)\n", watchpoint->id, watchpoint->segment, watchpoint->minAddress, watchpoint->segment, watchpoint->maxAddress);
```

Comment:
Even though this notation is _technically correct_, people who aren't familiar with bounded and unbounded interval notation will find this confusing. And I expect that will be a significant percentage of the users using this feature. Not to mention the fact that the segment is the same on both. Perhaps just `%02X:%X-%X`, even though it's kinda wishy washy.